CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/SyntaxNodeFile.swift

```diff
@@ -0,0 +1,359 @@
+                      }
+                    )
+                  )
+                )
+              }
+
+            ExprSyntax("self.init(data)")
+          }
+
+          if node.hasOptionalBaseTypeChild {
+            // TODO: Remove when we no longer support compiling in Swift 5.6. Change the
+            // above constructor to use `Optional<BaseType.none` instead.
+            try! InitializerDeclSyntax("""
+              /// This initializer exists solely because Swift 5.6 does not support
+              /// `Optional<ConcreteType>.none` as a default value of a generic parameter.
+              /// The above initializer thus defaults to `nil` instead, but that means it
+              /// is not actually callable when either not passing the defaulted parameter,
+              /// or passing `nil`.
+              ///
+              /// Hack around that limitation using this initializer, which takes a
+              /// `Missing*` syntax node instead. `Missing*` is used over the base type as
+              /// the base type would allow implicit conversion from a string literal,
+              /// which the above initializer doesn't support.
+              \(raw: node.generateInitializerDecl(optionalBaseAsMissing: true, currentIndentation: "  "))
+              """) {
+              FunctionCallExprSyntax(callee: ExprSyntax("self.init")) {
+                TupleExprElementSyntax(label: "leadingTrivia", expression: ExprSyntax("leadingTrivia"))
+                for child in node.children {
+                  if child.hasOptionalBaseType {
+                    TupleExprElementSyntax(label: child.swiftName, expression: ExprSyntax("Optional<\(raw: child.typeName)>.none"))
+                  } else if child.isUnexpectedNodes {
+                    TupleExprElementSyntax(expression: ExprSyntax("\(raw: child.swiftName)"))
+                  } else {
+                    TupleExprElementSyntax(label: child.swiftName, expression: ExprSyntax("\(raw: child.swiftName)"))
+                  }
+                }
+                TupleExprElementSyntax(label: "trailingTrivia", expression: ExprSyntax("trailingTrivia"))
+              }
+            }
+          }
+
+          for (index, child) in node.children.enumerated() {
+            // ===================
+            // Children properties
+            // ===================
+
+            let childType = child.nodeChoices.isEmpty ? child.typeName : child.name
+            let type = child.isOptional ? TypeSyntax("\(raw: childType)?") : TypeSyntax("\(raw: childType)")
+            let childDoc = child.description.map { dedented(string: $0) }.map { Trivia.docLineComment("/// \($0)") + .newline }
+
+            VariableDeclSyntax(
+              leadingTrivia: childDoc ?? [],
+              modifiers: [DeclModifierSyntax(name: .keyword(.public))],
+              name: PatternSyntax("\(raw: child.swiftName)"),
+              type: TypeAnnotationSyntax(type: type)) {
```

Comment:
It might be worth adding a new convenience initializer to `VariableDecl` that takes multiple `AccessorDeclSyntax` using a builder. Doesn’t have to be for this PR though.

Reply:
We can use the one already there 🥳